omnigibson/robots/manipulation_robot.py

```diff
@@ -322,37 +346,61 @@ def _infer_finger_properties(self):
                     finger_pts[:, 2] > (finger_parent_max_z + finger_range * m.MIN_AG_DEFAULT_GRASP_POINT_PROP)
                 )[0]
                 finger_pts = finger_pts[valid_idxs]
-                # Make sure all points lie on a single side of the EEF's y-axis
-                y_signs = th.sign(finger_pts[:, 1])
-                assert th.all(
-                    y_signs == y_signs[0]
-                ).item(), "Expected all finger points to lie on single side of EEF y-axis!"
-                y_sign = y_signs[0].item()
-                y_abs_min = th.abs(finger_pts[:, 1]).min().item()
+                # Infer which side of the gripper corresponds to the inner side (i.e.: the side that touches betwen the
+                # two fingers
+                # Because we don't know what state the robot is in, the gripper might be open, closed, or some
+                # intermediate state.
+                # Therefore, we need to automatically infer which side is the inner
+                # We use the heuristic that given a set of points defining a gripper finger, we assume that it must one
+                # of (y_min, y_max) over all points, with the selection being chosen by using
+                # argmin(abs(y_min), abs(y_max). This accounts for the edge case where a closed gripper can result
+                # in a few points overlapping the EEF frame in the y-direction, leading to us having to check values
+                # that may be either positive or negative
+                y_min, y_max = finger_pts[:, 1].min(), finger_pts[:, 1].max()
+                abs_y_min, abs_y_max = abs(y_min), abs(y_max)
+                y_offset = abs_y_min if abs_y_min < abs_y_max else abs_y_max
+                y_sign = 1.0 if abs_y_min < abs_y_max else -1.0
```

Comment:
hmm I am not sure this heuristics actually work.. can we discuss in person?

Reply:
update to check for relative y positions of each finger in order to infer which side is the interior for each finger

Reply:
done